{{cookiecutter.project_slug}}/.pre-commit-config.yaml

```
@@ -0,0 +1,32 @@
+exclude: .+/migrations/.+\.py
+default_stages: [commit]
+fail_fast: true
+
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: master
+    hooks:
+      - id: trailing-whitespace
+        files: (^|/){{cookiecutter.project_slug}}/.+\.(py|html|sh|css|js)$
+
+-   repo: local
+    hooks:
+      - id: black
+        name: black
+        entry: black
+        args: [--line-length=80]
```

This should probably match [what we test for](https://github.com/pydanny/cookiecutter-django/blob/19eb42a7cf1bdf41fa3deef0af976ba2c7f2d1d3/tests/test_docker.sh#L23):

```suggestion
        args: [--exclude 'migrations']
```

---

so, I will remove `black` in the end.

---

Sorry that I respond so slowly. Since the two options in that test don't write the files back, and I add `pre-commit` for the generated project, not for this project(namely cookiecutter-django itself), so that when submission occurs, `pre-commit` can automatically format the code in file.  Later, I will make another PR. @browniebroke